src/main/java/com/alvarium/annotators/AbstractAnnotator.java

```diff
@@ -34,6 +36,16 @@
  */
 abstract class AbstractAnnotator {
 
+  protected  Logger logger;
+
+  public Logger getLogger() {
+    return logger;
+  }
+
+  public void setLogger(Logger logger) {
+    this.logger = logger;
+  }
+  
```

Comment:
I moved setting the logger in the constructor for now and called the super in the custom annotators for now till we check the proposed Logger singleton.

Reply:
Also for the Singleton in d04b485, I think it'd be a better implementation, but then can't we avoid even storing a logger property in the SDK and use the singleton instead, eliminating this line. 
```java
 this.logger = LoggerSingleton.getInstance();
```
Not sure then if this will be the best place to instantiate the singleton then.